### DISCUSSION TITLE
Clarification : Subdomain blocking

### DISCUSSION THREAD

**Discussion by theapache64** [SelfControlApp/selfcontrol#754] [General]
So when I say block netflix.com, does it block all these subdomains? 🤔

https://github.com/SelfControlApp/selfcontrol/blob/7d350dab078c46f9861b135d195bb518713efdf7/Block%20Management/BlockManager.m#L332-L335

**Comment by cstigler** [comment]
Yep! If you don't want that behavior, turn off the "block common subdomains" setting

**Comment by theapache64** [comment]
Here's my settings

![image](https://user-images.githubusercontent.com/9678279/141695137-09b679b8-b50e-49c0-9589-c9f13d20ae9d.png)

and here's my block list

![image](https://user-images.githubusercontent.com/9678279/141695155-8b6f94d2-fa5b-4622-8749-510fe5a10a0d.png)

and here's my `/etc/host` file after the timer.

```
# BEGIN SELFCONTROL BLOCK
0.0.0.0	youtube.com
::	youtube.com
0.0.0.0	www.youtube.com
::	www.youtube.com
0.0.0.0	stumbleupon.com
::	stumbleupon.com
0.0.0.0	vine.co
::	vine.co
0.0.0.0	dvdplay.one
::	dvdplay.one
0.0.0.0	facebook.com
::	facebook.com
0.0.0.0	facebook.com
::	facebook.com
0.0.0.0	netflix.com
::	netflix.com
0.0.0.0	netflix.com
::	netflix.com
0.0.0.0	pinterest.com
::	pinterest.com
0.0.0.0	reddit.com
::	reddit.com
0.0.0.0	dailymotion.com
::	dailymotion.com
0.0.0.0	tumblr.com
::	tumblr.com
0.0.0.0	hulu.com
::	hulu.com
0.0.0.0	instagram.com
::	instagram.com
0.0.0.0	linkedin.com
::	linkedin.com
0.0.0.0	www.stumbleupon.com
::	www.stumbleupon.com
0.0.0.0	www.vine.co
::	www.vine.co
0.0.0.0	www.dvdplay.one
::	www.dvdplay.one
0.0.0.0	www.facebook.com
::	www.facebook.com
0.0.0.0	www.facebook.com
::	www.facebook.com
0.0.0.0	twitter.com
::	twitter.com
0.0.0.0	buzzfeed.com
::	buzzfeed.com
0.0.0.0	www.instagram.com
::	www.instagram.com
0.0.0.0	www.linkedin.com
::	www.linkedin.com
0.0.0.0	funnyordie.com
::	funnyordie.com
0.0.0.0	www.reddit.com
::	www.reddit.com
0.0.0.0	www.tumblr.com
::	www.tumblr.com
0.0.0.0	www.netflix.com
::	www.netflix.com
0.0.0.0	www.netflix.com
::	www.netflix.com
0.0.0.0	www.pinterest.com
::	www.pinterest.com
0.0.0.0	www.dailymotion.com
::	www.dailymotion.com
0.0.0.0	www.buzzfeed.com
::	www.buzzfeed.com
0.0.0.0	www.funnyordie.com
::	www.funnyordie.com
0.0.0.0	www.hulu.com
::	www.hulu.com
0.0.0.0	www.twitter.com
::	www.twitter.com
0.0.0.0	api.twitter.com
::	api.twitter.com
0.0.0.0	9gag.com
::	9gag.com
0.0.0.0	www.9gag.com
::	www.9gag.com
0.0.0.0	collegehumor.com
::	collegehumor.com
0.0.0.0	www.collegehumor.com
::	www.collegehumor.com
# END SELFCONTROL BLOCK
```

I've asked the SCA to block `twitter.com`, it also blocked `api.twitter.com` as expected. I've also asked to block `netflix.com` but it didn't block the netflix sub domains. any idea why?

**Comment by cstigler** [comment]
Hm, not sure on that one. But as a workaround, you can always just add those domains to your list manually.

**Comment by oscarcalderonc** [comment]
Just curious, is it possible to use wildcards? Like, I uncheck "Block common subdomains", but if for a certain domain I want to block the main domain and all subdomains, to set something like *.facebook.com instead of adding every subdomain for facebook.com
